doc/source/rllib/rllib-offline.rst

```diff
@@ -5,437 +5,2083 @@
+through a specified schema that organizes the expert data. The API default schema for reading data is provided in
+:py:data:`~ray.rllib.offline.offline_prelearner.SCHEMA`.
 
-.. code-block:: bash
+Lets consider a simple example in which your expert data is stored with the schema: ``(o_t, a_t, r_t, o_tp1, d_t, i_t, logprobs_t)``. In this case 
+you provide this schema as follows:
 
-    $ rllib train \
-        --run=PG \
-        --env=CartPole-v1 \
-        --config='{"output": "/tmp/cartpole-eval", "output_max_file_size": 5000000}' \
-        --stop='{"timesteps_total": 10000}'
+.. code-block:: python
 
-.. hint:: You should use separate datasets for algorithm training and OPE, as shown here.
+    from ray.rllib.algorithms.bc import BCConfig
+    from ray.rllib.core.columns import Columns
 
-We can now train a DQN algorithm offline and evaluate it using OPE:
+    config = (
+        BCConfig()
+        ...
+        .offline_data(
+            input_=[<input_path>],
+            # Provide the schema of your data.
+            input_read_schema={
+                Columns.OBS: "o_t",
+                Columns.ACTIONS: "a_t",
+                Columns.REWARDS: "r_t",
+                Columns.NEXT_OBS: "o_tp1",
+                Columns.INFOS: "i_t",
+                "done": "d_t",
+            },
+        )
+    )
+        
+.. note:: Internally, the legacy ``gym``'s ``done`` signals are mapped to ``gymnasium``'s ``terminated`` signals, with ``truncated`` values defaulting to 
+    ``False``. RLlib's :py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode` structures align with ``gymnasium``, adhering to the updated environment API standards in reinforcement learning.
 
-.. code-block:: python
+Converting Tabular Data to RLlib's Episode Format
+-------------------------------------------------
+While the tabular format is widely compatible and seamlessly integrates with RLlib's new Offline RL API, there are cases where you may prefer to use RLlib's native episode format. 
+As briefly mentioned earlier, such scenarios typically arise when full expert trajectories are required.
 
-    from ray.rllib.algorithms.dqn import DQNConfig
-    from ray.rllib.offline.estimators import (
-        ImportanceSampling,
-        WeightedImportanceSampling,
-        DirectMethod,
-        DoublyRobust,
+.. note::
+    RLlib processes tabular data in batches, converting each row into a *single-step episode*. This approach is primarily for procedural simplicity, as data can't
+    generally be assumed to arrive in time-ordered rows grouped by episodes, though this may occasionally be the case (however knowledge of such a structure resides  
+    with the user as RLlib can't easily infer it automatically). While it's possible to concatenate :py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode`
+    chunks, this can't be done with fragmented chunks.
```

Comment:
```suggestion
    with the user as RLlib can't easily infer it automatically). While it's possible to concatenate consecutive :py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode`
    chunks, this can't be done with chunks arriving in some scrambled order.
```